srcpkgs/mercury/template

```diff
@@ -0,0 +1,151 @@
+maintainer="Emily McDonough <emily@alaskanemily.net>"
+license="GPL-2.0-only, custom:LGPL-2.0-only-linking-exception"
+homepage="https://mercurylang.org/"
+distfiles="https://dl.mercurylang.org/release/mercury-srcdist-${version}.tar.xz"
+checksum=4ab0734aaa66808c91ffe7f54c7d818bdc4f0247aa255932e1e3d50610db4150
+nocross="Mercury cannot be cross-compiled to different architectures"
+noverifyrdeps=yes
+noshlibprovides=yes
+shlib_requires="libc.so.6"
+
+# asm_fast is preferable, but does not work on all architectures.
+case "$XBPS_MACHINE" in
+	*x86_64*|*arm*)
+		_LLDS_GRADE=asm_fast ;;
+	*i686*)
+		_LLDS_GRADE=reg ;;
+	*)
+		_LLDS_GRADE=none ;;
+esac
+
+_LIBGRADES="hlc.gc,hlc.gc.ssdebug,hlc.gc.memprof,${_LLDS_GRADE}.gc,${_LLDS_GRADE}.gc.stseg,${_LLDS_GRADE}.gc.decldebug.stseg,${_LLDS_GRADE}.gc.profdeep.stseg"
+
+# The parallel grades have some issues on musl systems.
+if [ "$XBPS_LIBC" != musl ] ; then
+	_LIBGRADES="${_LIBGRADES},hlc.par.gc,${_LLDS_GRADE}.par.gc.stseg"
+fi
+
+do_configure() {
+	# Prefer the low-level C grade as default, unless it is none.
+	if [ "${_LLDS_GRADE}" != "none" ] ; then
+		DEFAULT_GRADES="${_LLDS_GRADE}.gc.stseg"
+	else
+		DEFAULT_GRADES=hlc.gc
+	fi
+
+	./configure --prefix=/usr/ \
+		--enable-new-mercuryfile-struct \
+		--enable-gc-mmap --enable-gc-munmap \
+		--without-readline --with-editline \
+		--enable-libgrades="${_LIBGRADES}" --with-llds-base-grade="${_LLDS_GRADE}" --with-default-grade="${DEFAULT_GRADES}"
+}
+
+do_build() {
+	# The Mercury makefiles do most of the actual compilation in the 'install'
+	# rule. The 'DESTDIR' makefile variable is used to have the makefile
+	# install to the build directory instead, and then copy the files in
+	# do_install().
+	make PARALLEL="${makejobs}" DESTDIR=$(pwd)
+	make PARALLEL="${makejobs}" DESTDIR=$(pwd) install
+}
+
+do_install() {
+	vmkdir /usr/lib/mercury/lib
+	vmkdir /usr/lib/mercury/conf
+	vman usr/share/man/man1/c2init.1
+	vman usr/share/man/man1/mdb.1
+	vman usr/share/man/man1/mercury_config.1
+	vman usr/share/man/man1/mgnuc.1
+	vman usr/share/man/man1/ml.1
+	vman usr/share/man/man1/mmake.1
+	vman usr/share/man/man1/mmc.1
+	vman usr/share/man/man1/mtags.1
+	vbin usr/bin/c2init
+	vbin usr/bin/canonical_grade
+	vbin usr/bin/mdb
+	vbin usr/bin/mdemangle
+	vbin usr/bin/mercury_compile
+	vbin usr/bin/mercury_config
+	vbin usr/bin/mfiltercc
+	vbin usr/bin/mfilterjavac
+	vbin usr/bin/mgnuc
+	vbin usr/bin/mkinit
+	vbin usr/bin/ml
+	vbin usr/bin/mmake
+	vbin usr/bin/mmc
+	vbin usr/bin/mtags
+	vcopy usr/lib/mercury/conf/Mercury.config /usr/lib/mercury/conf/Mercury.config
+	vcopy usr/lib/mercury/conf/mercury_conf.h /usr/lib/mercury/conf/mercury_conf.h
+	vcopy usr/lib/mercury/ints /usr/lib/mercury/ints
+	vcopy usr/lib/mercury/inc /usr/lib/mercury/inc
+	vcopy usr/lib/mercury/ints /usr/lib/mercury/ints
+	vcopy usr/lib/mercury/mdb /usr/lib/mercury/mdb
+	vcopy usr/lib/mercury/mmake /usr/lib/mercury/mmake
+	vcopy usr/lib/mercury/lib/libgc*.a /usr/lib/mercury/lib
```

Comment:
also should not have leading slashes, i think

Reply:
Done